extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -847,6 +853,14 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
       this.fireDelayedImpressions(responseHeaders.get('X-AmpImps'));
     }
 
+    // If the response included a pageview state token, check for an existing
+    // token and remove it. Then save the new one to the module level object.
+    if (responseHeaders.get('_pvtoken_')) {
+      this.removePageviewStateToken();
```

Comment:
No though I can see why you'd expect it to. The tokens are keys in the object that stores them and the values are the slots associated with the token. So adding a new token will leave the old one in place and the object could get quite large.

I know this seems (and perhaps is) backwards but it would be more of a pain to try to use an instance as a key (and Keith didn't want me to add IDs to slots if we could avoid it).

Reply:
Got it, thanks!